app/editor/src/features/admin/reports/ReportForm.tsx

```diff
@@ -62,7 +63,18 @@ export const ReportForm: React.FC = () => {
   const handleSubmit = async (values: IReportModel) => {
     try {
       const originalId = values.id;
-      const result = !report.id ? await addReport(values) : await updateReport(values);
+      // Update event schedule information because names must be unique.
+      const report = {
+        ...values,
+        schedules: [
+          ...values.schedules.map((s, i) => ({
+            ...s,
+            name: generateScheduleName(`Schedule ${i + 1}`, values),
```

Comment:
I don't love this implementation, but the original goal was to keep schedules uniquely named to identify them.